packages/pillarbox-playlist/src/pillarbox-playlist-modal.js

```diff
@@ -0,0 +1,228 @@
+const PillarboxPlaylistMenuItem = videojs.getComponent('PillarboxPlaylistMenuItem');
+/**
+ * @ignore
+ * @type {typeof import('video.js/dist/types/button').default}
+ */
+const Button = videojs.getComponent('Button');
+/**
+ * @ignore
+ * @type {typeof import('video.js/dist/types/component').default}
+ */
+const Component = videojs.getComponent('Component');
+/**
+ * @ignore
+ * @type {typeof import('video.js/dist/types/modal-dialog').default}
+ */
+const ModalDialog = videojs.getComponent('ModalDialog');
+
+/**
+ * PlaylistMenuDialog is a custom dialog that extends the ModalDialog class.
+ * It is designed to manage and display a playlist with various controls.
+ */
+class PlaylistMenuDialog extends ModalDialog {
+  /**
+   * Handles the 'statechanged' event when triggered. This method serves as a proxy to
+   * the main `statechanged` handler, ensuring that additional logic can be executed or
+   * making it easier to detach the event listener later.
+   *
+   * @private
+   */
+  _onPlaylistStateChanged = ({ changes }) => {
+    if ('items' in changes) {
+      this.removeItems();
+      this.renderItems();
+    }
+
+    if ('currentIndex' in changes) {
+      this.select(changes.currentIndex.to);
+    }
+  };
+
+  /**
+   * Creates an instance of PlaylistMenuDialog.
+   *
+   * @param {import('@srgssr/pillarbox-web').Player} player - The pillarbox player instance.
+   * @param {Object} options - Options for the dialog.
+   */
+  constructor(player, options) {
+    options.temporary = false;
+
+    super(player, options);
+    this.fill();
+    this.renderComponent();
+    this.playlist().on('statechanged', this._onPlaylistStateChanged);
+  }
+
+  buildCSSClass() {
+    return `pbw-playlist-dialog ${super.buildCSSClass()}`;
+  }
+
+  /**
+   * Dispose of the PlaylistMenuDialog instance.
+   */
+  dispose() {
+    this.playlist().off('statechanged', this._onPlaylistStateChanged);
+    super.dispose();
+  }
+
+  /**
+   * Get the playlist instance associated with the player.
+   *
+   * @returns {import('pillarbox-playlist.js').default} The playlist instance.
+   */
+  playlist() {
+    return this.player().pillarboxPlaylist();
+  }
+
+  /**
+   * Update the playlist item UI with the selected index.
+   *
+   * @param {number} index - The index of the item to select.
+   */
+  select(index) {
+    this.children()
+      .filter(child => child.name() === 'PillarboxPlaylistMenuItem')
+      .forEach(child => child.selected(index === child.options().index));
+  }
+
+  /**
+   * Remove all playlist items from the dialog.
+   */
+  removeItems() {
+    this.children()
+      .filter(child => child.name() === 'PillarboxPlaylistMenuItem')
+      .forEach(child => this.removeChild(child));
+  }
+
+  /**
+   * Render the playlist items in the dialog.
+   */
+  renderItems() {
+    this.playlist().items.forEach((item, index) => this.addChild(
+      new PillarboxPlaylistMenuItem(this.player(), { item, index, name: 'PillarboxPlaylistMenuItem' })
+    ));
```

Comment:
Could we wrap everything in a `ol/li` and remove the `index`. This avoids having buttons flying around in the div.